pkg/volume/gce_pd/gce_util.go

```diff
@@ -86,33 +86,30 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 	// Apply Parameters (case-insensitive). We leave validation of
 	// the values to the cloud provider.
 	diskType := ""
-	zone := ""
+	var PutAdminAndUserRequestsTogetherParams volume.PutAdminAndUserRequestsTogetherParams
 	for k, v := range c.options.Parameters {
 		switch strings.ToLower(k) {
 		case "type":
 			diskType = v
 		case "zone":
-			zone = v
+			PutAdminAndUserRequestsTogetherParams.IsSCZoneSpecified = true
```

Comment:
The `c.options` are validated and lower-cased in the `func CreateVolume`. If we put `c.options` into `volume.PutAdminAndUserRequestsTogether` the `for k, v := range c.options.Parameters` will have to be repeated inside the `func` and the  `c.options.Parameters` will have to be lower-cased again.
Note: I wouldn't put the validation of the `c.options.Parameters` into the `func volume.PutAdminAndUserRequestsTogether` because the `func` deals with `zone, zones and PVC` only.
On the other hand if the `func volume.PutAdminAndUserRequestsTogether` is made a class it might get a little bit better.